anime_downloader/sites/4anime.py

```diff
@@ -0,0 +1,45 @@
+import logging
+from anime_downloader.sites.anime import Anime, AnimeEpisode, SearchResult
+from anime_downloader.sites import helpers
+
+logger = logging.getLogger(__name__)
+
+class Anime4(Anime, sitename = '4anime'):
+    sitename = '4anime'
+
+    @classmethod
+    def search(cls, query):
+        data = {
+            "action": "ajaxsearchlite_search", 
+            "aslp": query, 
+            "asid": 1, 
+            "options": "qtranslate_lang=0&set_intitle=None&customset%5B%5D=anime"
+            }
+        soup = helpers.soupify(helpers.post("https://4anime.to/wp-admin/admin-ajax.php", data=data).text)
+
+        search_results = [
+            SearchResult(
+                title = i.find('div', class_='info').a.text,
+                url = i.find('div', class_='info').a['href']
+                )
+            for i in soup.find_all('div', class_='item')
+            ]
+        return search_results
+
+    def _scrape_episodes(self):
+        soup = helpers.soupify(helpers.get(self.url).text).find('ul', class_='episodes range active').find_all('li')
+        return [x.a['href'] for x in soup]
+
+    def _scrape_metadata(self):
+        soup = helpers.soupify(helpers.get(self.url).text)
+        self.title = soup.title.text
+        for item in soup.find_all('div', class_='detail'):
+            more = item.find_all('a')
+            for i in more:
+                if 'year' in i['href']:
+                    self.year = int(i.text)

```

Comment:
honestly no idea if this is a standard or should be done líke this

Reply:
it was the only way i could get the year without using an index number that would be prone to errors

Reply:
it might be an overkill just because of the Lines Of Code it consumes, but it only iterates over 5 or 6 items and all the links in each of item, not much to fear for slowdown

Reply:
No faults in the code, just unsure if this is how to correctly define metadata in this project

Reply:
by define you meant how i set self.year?

Reply:
yes, it could be something like self.metadata['year'] = 2020 for all I know.

Reply:
> yes, it could be something like self.metadata['year'] = 2020 for all I know.
if we decide on a standard for the metadata in __scrape_metadata then we can easily change how it gets saved as metadata